lib/mongoid/criteria/queryable/extensions/symbol.rb

```diff
@@ -59,7 +59,9 @@ def add_key(name, strategy, operator, additional = nil, &block)
             #
             # @return [ Symbol ] The value as a symbol.
             def evolve(object)
-              __evolve__(object) { |obj| obj.to_sym }
+              __evolve__(object) do |obj|
+                obj.try(:to_sym)
```

Comment:
The `try` call is the same as `obj&.to_sym` right?

Reply:
I don't think so... iirc the `&.` safe-navigation operator either calls the method if the value is not nil, or returns nil if it is.

> What I'm trying to protect against here is the case where the to_sym method doesn't exist on the value (for example, arrays). Changing to the safe navigation operator should cause that case to raise a NoMethodError

Reply:
Aren't they the same?

```irb(main):001:0> [].try(:to_sym)
(irb):1:in `<main>': undefined method `try' for []:Array (NoMethodError)
	from /usr/lib/ruby/gems/3.0.0/gems/irb-1.3.5/exe/irb:11:in `<top (required)>'
	from /usr/bin/irb:23:in `load'
	from /usr/bin/irb:23:in `<main>'

```

Reply:
Try this again with rails loaded:
```
irb(main):001:0> [].try(:to_sym)
=> nil
irb(main):002:0> []&.to_sym
Traceback (most recent call last):
       16: from /Users/neil.shweky/.gem/gems/bundler-2.3.6/lib/bundler/friendly_errors.rb:103:in `with_friendly_errors'
       15: from /Users/neil.shweky/.gem/gems/bundler-2.3.6/exe/bundle:48:in `block in <top (required)>'
       14: from /Users/neil.shweky/.gem/gems/bundler-2.3.6/lib/bundler/cli.rb:25:in `start'
       13: from /Users/neil.shweky/.gem/gems/bundler-2.3.6/lib/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
       12: from /Users/neil.shweky/.gem/gems/bundler-2.3.6/lib/bundler/cli.rb:31:in `dispatch'
       11: from /Users/neil.shweky/.gem/gems/bundler-2.3.6/lib/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
       10: from /Users/neil.shweky/.gem/gems/bundler-2.3.6/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
        9: from /Users/neil.shweky/.gem/gems/bundler-2.3.6/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
        8: from /Users/neil.shweky/.gem/gems/bundler-2.3.6/lib/bundler/cli.rb:484:in `exec'
        7: from /Users/neil.shweky/.gem/gems/bundler-2.3.6/lib/bundler/cli/exec.rb:23:in `run'
        6: from /Users/neil.shweky/.gem/gems/bundler-2.3.6/lib/bundler/cli/exec.rb:58:in `kernel_load'
        5: from /Users/neil.shweky/.gem/gems/bundler-2.3.6/lib/bundler/cli/exec.rb:58:in `load'
        4: from /Users/neil.shweky/.gem/bin/irb:23:in `<top (required)>'
        3: from /Users/neil.shweky/.gem/bin/irb:23:in `load'
        2: from /Users/neil.shweky/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/irb-1.2.6/exe/irb:11:in `<top (required)>'
        1: from (irb):2
NoMethodError (undefined method `to_sym' for []:Array)
Did you mean?  to_s
               to_set
```

Reply:
I see.